src/components/Details/Flavor/Flavor.js

```diff
@@ -39,17 +40,28 @@ export class Flavor extends React.Component {
       const getTemplatePromise = this.props.k8sGet(TemplateModel, template.name, template.namespace);
       getTemplatePromise
         .then(result => {
```

Comment:
I think this would be better to move to another file and get templates here in one final promise with loadError in catch clause